src/TesApi.Web/Management/Batch/TerraBatchPoolManager.cs

```diff
@@ -146,6 +144,28 @@ public async Task DeleteBatchPoolAsync(string poolId, CancellationToken cancella
             }
         }
 
+        private async Task<Guid> GetWsmResourceIdFromBatchPoolMetadataAsync(string poolId, CancellationToken cancellationToken)
+        {
+            var batchClient = CreateBatchClientFromOptions();
+
+            var pool = await batchClient.PoolOperations.GetPoolAsync(poolId, cancellationToken: cancellationToken);
+
+            if (pool is null)
+            {
+                throw new InvalidOperationException($"The batch pool was not found. Pool id:{poolId}");
+            }
+
+            var metadataItem = pool.Metadata.SingleOrDefault(m => m.Name.Equals(TerraResourceIdMetadataKey));
```

Comment:
should add OrdinalIgnoreCase to this

Reply:
`TerraResourceIdMetadataKey` is a constant and used in when setting and reading. The case should be exact, no need to assume other.

Reply:
it's faster to use `StringComparison.OrdinalIgnoreCase` if we can! :-)

Reply:
It's actually both faster and more reliable to add either `StringComparison.OrdinalIgnoreCase` or `StringComparison.Ordinal` than to leave it bare. Leaving it bare causes locale rules based on current culture settings to both slow things down and to sometimes produce different results.